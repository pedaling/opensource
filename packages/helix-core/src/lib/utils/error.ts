import type { Class } from '@helix-js/utils';

export class OptimisticConcurrencyError extends Error {}

export async function retryIfError<TReturn>(
  logicToRetry: () => Promise<TReturn>,
  targetErrorClass: Class<Error>,
  maxRetries = 1000
): Promise<TReturn> {
  let errorAfterMaxTries: Error | undefined;

  for (let tryNumber = 1; tryNumber <= maxRetries; tryNumber++) {
    try {
      const result = await logicToRetry();

      return result;
    } catch (e) {
      const error = e as Error;

      checkRetryError(error, targetErrorClass);

      errorAfterMaxTries = error;
    }
  }

  throw new Error(
    `[retryIfError] Reached the maximum number of retries (${maxRetries}), but still getting the following error: ${errorAfterMaxTries}`
  );
}

function checkRetryError(error: Error, errorClassThatRetries: Class<Error>): void {
  if (!(error instanceof errorClassThatRetries)) {
    throw error;
  }
}
