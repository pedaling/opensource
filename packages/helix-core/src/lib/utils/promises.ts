export class Promises {
  public static async allSettledAndFulfilled<TValue>(
    values: Iterable<TValue>
  ): ReturnType<PromiseConstructor['allSettled']> {
    const results = await Promise.allSettled(values);

    const failed: PromiseRejectedResult[] = results
      .filter(res => res.status == 'rejected')
      .map(res => res as PromiseRejectedResult);

    if (failed.length > 0) {
      throw new PromisesError(failed);
    }

    return results;
  }
}

export class PromisesError extends Error {
  public readonly failedReasons: unknown[];

  constructor(rejectedResults: PromiseRejectedResult[]) {
    const reasons = rejectedResults.map(res => res.reason);

    super(reasons.join('. '));

    this.failedReasons = reasons;
  }
}
