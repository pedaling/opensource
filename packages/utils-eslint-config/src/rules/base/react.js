module.exports = {
  plugins: ['react'],
  rules: {
    'react/jsx-boolean-value': ['error', 'always'],
    'react/jsx-key': 'error',
    'react/jsx-props-no-spreading': 'off',
    'react/jsx-uses-react': 'off',
    'react/jsx-wrap-multilines': 0,
    'react/no-array-index-key': 1,
    'react/no-unused-prop-types': 0,
    'react/prop-types': 'off',
    'react/react-in-jsx-scope': 'off',
    'react/require-default-props': 0,
    'react/sort-comp': 0,
    'react/state-in-constructor': 0,
    'react/static-property-placement': 0,
    'react/jsx-curly-brace-presence': ['error', { props: 'never', children: 'never' }],

    // https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-no-useless-fragment.md
    'react/jsx-no-useless-fragment': ['error', { allowExpressions: true }],

    'react/function-component-definition': 'off',
  },
};
