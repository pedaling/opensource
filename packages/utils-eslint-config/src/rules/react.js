const reactRules = {
  'react/function-component-definition': ['off'],
  'react/jsx-boolean-value': ['error', 'always'],
  'react/jsx-curly-brace-presence': ['error', { props: 'never', children: 'never' }],
  'react/jsx-key': ['error'],
  'react/jsx-no-useless-fragment': ['error', { allowExpressions: true }],
  'react/jsx-props-no-spreading': ['off'],
  'react/jsx-uses-react': ['off'],
  'react/jsx-wrap-multilines': ['off'],
  'react/no-array-index-key': ['off'],
  'react/no-unused-prop-types': ['off'],
  'react/prop-types': ['off'],
  'react/react-in-jsx-scope': ['off'],
  'react/require-default-props': ['off'],
  'react/sort-comp': ['off'],
  'react/state-in-constructor': ['off'],
  'react/static-property-placement': ['off'],
  'react/self-closing-comp': ['error'],
};

module.exports = reactRules;
