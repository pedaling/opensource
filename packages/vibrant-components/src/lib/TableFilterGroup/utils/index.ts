import type { DateFilterOperator, MultiSelectFilterOperator, Option, StringFilterOperator } from '../types';

export const isValueRequiredOperator = (op: string) => op === 'empty' || op === 'notEmpty';

export const isStringFilterValid = (filter: { value: string; operator: StringFilterOperator }) =>
  Boolean(filter.value) || isValueRequiredOperator(filter.operator);

export const isDateFilterValid = (filter: { value: Date[]; operator: DateFilterOperator }) => {
  if (isValueRequiredOperator(filter.operator)) return true;

  if (filter.operator === 'between') return filter.value.length >= 2;

  return filter.value.length >= 1;
};

export const isMultiSelectFilterValid = (filter: { value: Option['value'][]; operator: MultiSelectFilterOperator }) =>
  Boolean(filter.value.length > 0) || isValueRequiredOperator(filter.operator);
