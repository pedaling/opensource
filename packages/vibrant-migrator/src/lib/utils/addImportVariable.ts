import type { API, ASTPath, ImportDeclaration } from 'jscodeshift';

type AddImportVariableParams = {
  importPath: ASTPath<ImportDeclaration>;
  variableName: string;
  packageName: string;
  jscodeshift: API['jscodeshift'];
};

export function addImportVariable({ importPath, variableName, packageName, jscodeshift }: AddImportVariableParams) {
  try {
    jscodeshift(importPath).insertAfter(
      jscodeshift.importDeclaration(
        [jscodeshift.importSpecifier(jscodeshift.identifier(variableName))],
        jscodeshift.literal(packageName)
      )
    );
    // eslint-disable-next-line no-empty
  } catch {}
}
