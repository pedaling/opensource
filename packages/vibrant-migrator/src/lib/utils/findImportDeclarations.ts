import type { API, Collection } from 'jscodeshift';
import { ImportDeclaration } from 'jscodeshift';

type FindImportDeclarationsParams = {
  root: Collection;
  importPackage: string;
  jscodeshift: API['jscodeshift'];
};

export function findImportDeclarations({ root, importPackage }: FindImportDeclarationsParams) {
  return root.find(ImportDeclaration, node => node.type === 'ImportDeclaration' && node.source.value === importPackage);
}
