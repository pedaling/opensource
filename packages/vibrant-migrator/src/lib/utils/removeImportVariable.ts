import type { API, ASTPath, ImportDeclaration } from 'jscodeshift';

type RemoveImportVariableParams = {
  importPath: ASTPath<ImportDeclaration>;
  variableName: string;
  jscodeshift: API['jscodeshift'];
};

export function removeImportVariable({ importPath, variableName, jscodeshift }: RemoveImportVariableParams) {
  const importDeclaration = importPath.node;
  const importVariables = (importDeclaration.specifiers ?? []).filter(
    specifier => specifier.type !== 'ImportSpecifier' || specifier.imported.name !== variableName
  );

  if (importVariables.length === 0) {
    jscodeshift(importDeclaration.source).remove();

    return;
  }

  jscodeshift(importPath).replaceWith(jscodeshift.importDeclaration(importVariables, importDeclaration.source));
}
