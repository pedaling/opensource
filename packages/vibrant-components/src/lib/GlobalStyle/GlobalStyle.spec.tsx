import { waitFor } from '@testing-library/react';
import { GlobalStyle } from '@vibrant-ui/components';
import { createReactRenderer } from '@vibrant-ui/utils';

describe('<GlobalStyle />', () => {
  const { render } = createReactRenderer();

  beforeEach(() => {
    render(<GlobalStyle />);
  });

  it('should rendered', async () => {
    await waitFor(() =>
      expect(document.head.querySelectorAll('style[data-emotion="emotion-global"]').length).toBeGreaterThan(0)
    );
  });

  it('match snapshot', async () => {
    await waitFor(() =>
      expect(document.head.querySelectorAll('style[data-emotion="emotion-global"]')).toMatchSnapshot()
    );
  });
});
