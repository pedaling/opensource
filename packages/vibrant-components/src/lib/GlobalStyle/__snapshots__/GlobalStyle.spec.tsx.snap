// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GlobalStyle /> match snapshot 1`] = `
NodeList [
  <style
    data-emotion="emotion-global"
    data-s=""
  >
    
    html,body,p,ol,ul,li,dl,dt,dd,blockquote,figure,fieldset,legend,textarea,pre,iframe,hr,h1,h2,h3,h4,h5,h6{margin:0;padding:0;}
  </style>,
  <style
    data-emotion="emotion-global"
    data-s=""
  >
    
    h1,h2,h3,h4,h5,h6{font-size:100%;font-weight:normal;}
  </style>,
  <style
    data-emotion="emotion-global"
    data-s=""
  >
    
    ul,li{list-style:none;}
  </style>,
  <style
    data-emotion="emotion-global"
    data-s=""
  >
    
    button,input,select{margin:0;}
  </style>,
  <style
    data-emotion="emotion-global"
    data-s=""
  >
    
    html{box-sizing:border-box;}
  </style>,
  <style
    data-emotion="emotion-global"
    data-s=""
  >
    
    *,*::before,*::after{box-sizing:inherit;}
  </style>,
  <style
    data-emotion="emotion-global"
    data-s=""
  >
    
    img,video{height:auto;max-width:100%;}
  </style>,
  <style
    data-emotion="emotion-global"
    data-s=""
  >
    
    iframe{border:0;}
  </style>,
  <style
    data-emotion="emotion-global"
    data-s=""
  >
    
    table{border-collapse:collapse;border-spacing:0;}
  </style>,
  <style
    data-emotion="emotion-global"
    data-s=""
  >
    
    td,th{padding:0;}
  </style>,
]
`;
